File: 1000/2093C/main.java
import java.util.*;
public class main
{
    public static boolean isPrime(long n)
    {
        for(int i=2;i<=Math.sqrt(n);i++)
        {
            if(n%i == 0)
                return false;
        }

        return true;
    }
 
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long n = sc.nextLong();
            int k = sc.nextInt();
            
            boolean prime = isPrime(n);
            
            if(prime == false)
                System.out.println("NO");
            else
            {
                if(n == 1)
                {
                    if(k == 2)
                        System.out.println("YES");
                    else
                        System.out.println("NO");
                }
                else
                {
                    if(k == 1)
                        System.out.println("YES");
                    else
                        System.out.println("NO");
                }
            }
        }
    }
}

File: 1000/2093C/main.cpp
#include <bits/stdc++.h>
using namespace std;

bool isPrime(long long n) 
{
    if (n < 2) 
        return false;

    for (long long i = 2; i * i <= n; i++) 
    {
        if (n % i == 0)
            return false;
    }

    return true;
}

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        long long n;
        int k;
        cin >> n >> k;

        bool prime = isPrime(n);

        if (!prime)
            cout << "NO\n";
        else 
        {
            if (n == 1) 
            {
                if (k == 2)
                    cout << "YES\n";
                else
                    cout << "NO\n";
            } 
            else
            {
                if (k == 1)
                    cout << "YES\n";
                else
                    cout << "NO\n";
            }
        }
    }

    return 0;
}
